Formula/b/build2.rb

```diff
@@ -0,0 +1,143 @@
+  # .../MacOS*.sdk/usr/lib/ directories (for example, libsqlite3.tbd). Drop
+  # them once upgraded to 0.17.0 which includes them.
+  #
+  patch do
+    url "https://github.com/build2/build2/commit/f221ca94a304abaa79239cf2cbaf3ffafbbc092c.patch?full_index=1"
+    sha256 "391fbd38e863bc3c23f1cddf360f097d210b31ad1e9a906eb583d927f0d216a3"
+    directory "build2"
+  end
+
+  patch do
+    url "https://github.com/build2/build2/commit/17c1214943959b304ab5cdcccacec8a70c7a6c0f.patch?full_index=1"
+    sha256 "a860ce05f75bf653a135c443fab8299e7c0c7e0980cb0f86583a37ba72ecce7c"
+    directory "build2"
+  end
+
+  def install
+    pkgconf = Formula["pkgconf"]
+    sqlite = Formula["sqlite"] if OS.linux?
+
+    # Note that the below steps are equivalent to invoking build.sh with the
+    # --local --private --system libpkgconf,libsqlite3 options. The only
+    # difference are a few customizations to find Homebrew's libpkgconf and
+    # libsqlite3.
+    #
+    # For background on these steps, see:
+    #
+    # https://build2.org/build2-toolchain/doc/build2-toolchain-install.xhtml
+
+    # Suppress loading of default options files.
+    #
+    ENV["BUILD2_DEF_OPT"] = "0"
+
+    # Bootstrap, stage 1.
+    #
+    # Note that we disable all warnings since we cannot do anything more
+    # granular during bootstrap stage 1.
+    #
+    cd "build2" do
+      system "make",
+             "-j", ENV.make_jobs,
+             "-f", "./bootstrap.gmake",
+             "CXX=#{ENV.cxx}", "CXXFLAGS=-w"
+    end
+
+    # Bootstrap, stage 2.
+    #
+    # Force using Homebrew's libpkgconf rather than building libpkg-config
+    # from source.
+    #
+    # Note that the build2 driver bootstrapped at stage 1 above doesn't read
+    # .pc files. Thus, during the bootstrap stage 2, it is assumed that the
+    # headers and libraries of the system dependencies can be found by the
+    # compiler and linker automatically. Failed that we have to manually
+    # specify their header/library paths, which is what we do here for
+    # libpkgconf.
+    #
+    cd "build2" do
+      system "build2/b-boot", "-v",
+             "-j", ENV.make_jobs,
+             "config.cxx=#{ENV.cxx}",
+             "config.bin.lib=static",
+             "config.build2.libpkgconf=true",
+             "config.cc.poptions=-I#{pkgconf.include}/pkgconf",
+             "config.cc.loptions=-L#{pkgconf.lib}",
+             "build2/exe{b}"
+      mv "build2/b", "build2/b-boot"
+    end
+
+    # Local build.
+    #
+    # Note that we don't build hermetically to avoid baking Homebrew-specific
+    # paths into the final dylibs.
+    #
+    loptions = "-L#{pkgconf.lib}"
+    loptions += " -L#{sqlite.lib}" if OS.linux?
+
+    system "build2/build2/b-boot", "-V",
+           "config.cxx=#{ENV.cxx}",
+           "config.cc.coptions=-O3",
+           "config.cc.loptions=#{loptions}",
+           "config.bin.lib=shared",
+           "config.bin.rpath=#{lib}/build2",
```

Comment:
```suggestion
           "config.bin.rpath=#{rpath}",
```

Reply:
I don't believe this will work unless we also get rid of the private installation (`config.install.private=build2`). Here is what happens: `build2` consists of a bunch of `.dylib` libraries which are build system modules. These libraries are not meant to be linked by other applications and not to pollute `$prefix/lib/` we install them into `$prefix/lib/build2/` and arrange for the corresponding `rpath`. This is also how our standard installation script installs things.

I can change that (i.e., install all the libraries directly into `$prefix/lib/`) but I don't see any benefits in doing that. The current layout works without any issues.

Reply:
Ok, I see what `#{rpath}` tries to accomplish (`@loader_path/../lib`). Let me see how I can work it in.

Reply:
Fixed in https://github.com/Homebrew/homebrew-core/pull/141550/commits/a97e5b511a89af6531e68b69f15e9ae6ea61d2bf , thanks for pointing this out!